documentation/development-docs/design-guidelines/managed-identity-best-practices.md

```diff
@@ -58,3 +58,35 @@ We are recommended to use string array as the type of UserAssignedIdentity with
 - No syntax changes if service supports one more user assigned identity in future;
 - Service will provide correct error response if customer reaches the count limitation of `UserAssignedIdentity` ideally, which means no harm.
 
+### How to disable transforming IdentityType and UserAssignedIdentity to avoid breaking changes when migrate from autorest.powershell v3 to v4.
```

Comment:
I did some refactorizes as below.

### What should I do to mitigate one patch operation which is reported to unable to be replaced by Get+Put operation.

autorest.powershell is unable to replace `patch` operation by `get` + `put` sometimes automatically for several reasons. To mitigate this issue,

- Include a customization script to support `get` + `put` update for this type of operations. The following are the detailed steps on how to accomplish this.
  - hide the corresponding Update cmdlet in directive by
  ```
   -  where:
        verb: Update
        subject: {Subject-Name}
      hide: true
  ```
  - run `autorest` and `./build-module.ps1`
  - copy internal/Update-Az{ModuleName}{Subject-Name}.ps1 to custom folder
  - manually change IdentityType to EnableSystemAssignedIdentity<bool> and UserAssignedIdentity<hashtable> to UserAssignedIdentity<string[]> like
  ```
      [Parameter()]
      [Microsoft.Azure.PowerShell.Cmdlets.Fleet.Category('Body')]
      [System.Nullable[System.Boolean]]
      # Decides if enable a system assigned identity for the resource.
      ${EnableSystemAssignedIdentity},
      [Parameter()]
      [AllowEmptyCollection()]
      [Microsoft.Azure.PowerShell.Cmdlets.Fleet.Category('Body')]
      [System.String[]]
      # The array of user assigned identities associated with the resource.
      # The elements in array will be ARM resource ids in the form: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}.'
      ${UserAssignedIdentity},
  ```
  - calculate the value of IdentityType and UserAssignedIdentity as swagger defined in process block, see example [here](https://github.com/Azure/azure-powershell/blob/2d37985acc5ca72f8838df884bd6b2b3abff3ae7/src/Fleet/Fleet.Autorest/generated/cmdlets/UpdateAzFleet_UpdateExpanded.cs#L445).
- After completing step one, you can suppress the errors by adding `suppress-replace-patch-by-getput-error: true` to the README.md file.